packages/js/product-editor/src/components/variations-table/use-variations/use-variations.ts

```diff
@@ -340,6 +340,9 @@ export function useVariations( { productId }: UseVariationsProps ) {
 		const { batchUpdateProductVariations, invalidateResolutionForStore } =
 			dispatch( EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME );
 
+		selectedVariationsRef.current = {};
+		setSelectedCount( 0 );
```

Comment:
I understand resetting the selectedVariationsRef, but is it needed to explicitly set the selected count to 0, if we are explicitly setting it within the for loop already?

Reply:
As you can see I'm incrementing the count within the loop, if I don't reset this here to 0 then the final amount could mismatch with the selected variations.